### PR TITLE
Add Joyent's Triton as a deploy target

### PR DESCRIPTION
**Pull request by BrianHicks**
- [x] Installs cleanly on a fresh build of most recent master branch
- [x] Updates documentation relevant to the changes

Upgrading is not applicable, this is a new target.

### CODE REVIEW

File: docs/getting_started/triton.rst [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+   variable key_path { default = "~/.ssh/id_rsa.pub" }
+
+Copy ``terraform/triton.sample.tf`` in it's entierty to the root of the project
+as ``triton.tf`` to start customization. In the next section, we'll explain how
+to obtain the settings mentioned above.
+
+Basic Settings
+^^^^^^^^^^^^^^
+
+First, we'll need an account. This is the username you use to log into Triton.
```
Comment (misterbisson) [inline docs/getting_started/triton.rst]: Possibly link to https://docs.joyent.com/public-cloud/getting-started to give people direction about how to create an account?

File: terraform/triton.sample.tf [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  image           = "${var.image}"
+  keys            = "${file(var.key_path)}"
+  package         = "${var.control_package}"
+  role            = "control"
+  short_name      = "${var.short_name}"
+  public_network  = "${var.public_network}"
+  private_network = "${module.network.network_id}"
+}
+
+module "worker-nodes" {
```
Comment (ryane) [inline terraform/triton.sample.tf]: is there any way to configure disk sizes for the different node types?

File: docs/getting_started/triton.rst [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+   }
+
+Provisioning
+------------
+
+Once your provider is set up, customize your modules (mainly for the variables
+ending in ``_count`` to control scaling.) Run ``terraform get`` to prepare the
+modules, ``terraform plan`` to see what will be created, and ``terraform apply``
+to provision the clsuter. Afterwards, you can use the instructions in
+:doc:`getting started <index>` to install Mantl on your new cluster.
```
Comment (misterbisson) [inline docs/getting_started/triton.rst]: Maybe add:

> **Note:** new Joyent accounts may be subject to provisioning limits. [Contact Joyent support](https://docs.joyent.com/public-cloud/getting-started) to have those limits raised.

File: docs/getting_started/triton.rst [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+       url = "https://us-east-1.api.joyentcloud.com"
+   }
+
+Provisioning
+------------
+
+Once your provider is set up, customize your modules (mainly for the variables
+ending in ``_count`` to control scaling.) Run ``terraform get`` to prepare the
+modules, ``terraform plan`` to see what will be created, and ``terraform apply``
+to provision the clsuter. Afterwards, you can use the instructions in
```
Comment (misterbisson) [inline docs/getting_started/triton.rst]: spelling: clsuter

File: playbooks/host-connectivity.yml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+---
```
Comment (ryane) [inline playbooks/host-connectivity.yml]: what is this used for? I know from discussion in slack but should we document it somewhere (even if it's just a comment in the playbook itself)?

Comment (BrianHicks) [inline playbooks/host-connectivity.yml]: done in 	d945e2e

File: docs/getting_started/triton.rst (hunk 1 above)
Comment (BrianHicks) [inline docs/getting_started/triton.rst]: done in 5cccc25

File: terraform/triton.sample.tf (hunk 2 above)
Comment (BrianHicks) [inline terraform/triton.sample.tf]: Not independently, it's the package argument.

**Comment by ryane** [comment]
lgtm (but needs rebased)

**Comment by BrianHicks** [comment]
Rebased and merged. Thanks for your review, @ryane and @misterbisson

**Comment by misterbisson** [comment]
Appears to be published at http://docs.mantl.io/en/latest/getting_started/triton.html

**Pull request merged**